monkey/tests/unit_tests/infection_monkey/utils/test_commands.py

```diff
@@ -0,0 +1,111 @@
+from infection_monkey.utils.commands import (
+    build_monkey_commandline_explicitly,
+    get_monkey_cmd_lines_linux,
+    get_monkey_cmd_lines_windows,
+)
+
+
+def test_build_monkey_commandline_explicitly():
+    test1 = [
+        "-p",
+        "101010",
+        "-t",
+        "10.10.101.10",
+        "-s",
+        "127.127.127.127:5000",
+        "-d",
+        "0",
+        "-l",
+        "C:\\windows\\abc",
+        "-vp",
+        "80",
+    ]
+    result1 = build_monkey_commandline_explicitly(
+        101010, "10.10.101.10", "127.127.127.127:5000", 0, "C:\\windows\\abc", 80
+    )
+
+    test2 = [
+        "-p",
+        "101010",
+        "-t",
+        "10.10.101.10",
+        "-s",
+        "200.150.100.50:5000",
+        "-d",
+        "0",
+        "-l",
+        "C:\\windows\\abc",
+        "-vp",
+        "443",
+    ]
+    result2 = build_monkey_commandline_explicitly(
+        101010, "10.10.101.10", "200.150.100.50:5000", -50, "C:\\windows\\abc", 443
+    )
```

Comment:
Try and keep tests as focused as possible and only test one thing. In this case, I believe we want to test that if we pass a negative number for the depth, it gets changed to a zero. If that's the case, all of the other parameters being passed obscure the intent of this test.